crates/mysten-common/src/notify_once.rs

```diff
@@ -18,6 +18,7 @@ use tokio::sync::Notify;
 ///
 /// NotifyOnce guarantees that wait() will return once notify() is called,
 /// regardless of whether wait() was called before or after notify().
+#[derive(Debug)]
```

Comment:
Curious why do we need derive(Debug) on it? (what useful information can we possible get from it)

Reply:
This was to satisfy DagError, because we are wrapping NotifyOnce in DagError::Suspended.

Reply:
We can implement a more useful Debug for whether the object is notified or not, if you prefer.

Reply:
I see, I guess we can keep this as is(the right fix is probably better design for dag interface, where we don't return an debug printable Error like this, but this is clearly not related to this PR)